### PR TITLE
Skip converts na values to strings

### LINKED ISSUE #44156: BUG: `astype("str")` converts NA values to strings
### 

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.
- [X] I have confirmed this bug exists on the master branch of pandas.

### Reproducible Example

```python
import pandas as pd
import numpy as np

pd.Series(["foo", "bar", pd.NA]).astype("str") // ["foo", "bar", "<NA>"]
pd.Series(["foo", "bar", np.NaN]).astype("str") // ["foo", "bar", "nan"]
pd.Series(["foo", "bar", None]).astype("str") // ["foo", "bar", "None"]
```

### Issue Description

NA/null values are converted to their string representation when doing `.astype("str")`, which seems like odd/undesirable behaviour.

### Expected Behavior

```
import pandas as pd
import numpy as np

pd.Series(["foo", "bar", pd.NA]).astype("str") // ["foo", "bar", pd.NA]
pd.Series(["foo", "bar", np.NaN]).astype("str") // ["foo", "bar", np.NaN]
pd.Series(["foo", "bar", None]).astype("str") // ["foo", "bar", None]
```

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 9b2bb732f09d8e870d806b337b97d3d99fd5b118
python           : 3.9.7.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 20.6.0
Version          : Darwin Kernel Version 20.6.0: Wed Jun 23 00:26:31 PDT 2021; root:xnu-7195.141.2~5/RELEASE_X86_64
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : en_GB.UTF-8
LOCALE           : en_GB.UTF-8

pandas           : 1.4.0.dev0+952.g9b2bb732f0
numpy            : 1.21.3
pytz             : 2021.3
dateutil         : 2.8.2
pip              : 21.3
setuptools       : 58.2.0
Cython           : 0.29.24
pytest           : 6.2.5
hypothesis       : 6.23.4
sphinx           : 4.2.0
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.6.3
html5lib         : 1.1
pymysql          : None
psycopg2         : None
jinja2           : 3.0.2
IPython          : None
pandas_datareader: None
bs4              : 4.10.0
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : 3.0.9
pandas_gbq       : None
pyarrow          : 5.0.0
pyxlsb           : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
tabulate         : 0.8.9
xarray           : None
xlrd             : 2.0.1
xlwt             : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by weikhor**
- [✓] closes #44156 
- [✓] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [✓] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/dtypes/test_dtypes.py [hunk 1]
```diff
@@ -1107,3 +1107,15 @@ def test_compare_complex_dtypes():
 
     with pytest.raises(TypeError, match=msg):
         df.lt(df.astype(object))
+
+
+@pytest.mark.parametrize(
+    "array",
+    [(["foo", "bar", pd.NA]), (["foo", "bar", np.NaN]), (["foo", "bar", None])],
+)
+def test_skip_convert_nan_values_to_strings(self, array):
```
Comment (phofl) [inline pandas/tests/dtypes/test_dtypes.py:1116]: You have to remove self, this is not in a class

Comment (phofl) [reply pandas/tests/dtypes/test_dtypes.py:1116]: Also this broke other tests

Comment (phofl) [reply pandas/tests/dtypes/test_dtypes.py]: You have to fix the other tests

**Comment by weikhor** [comment]
@phofl Hi This issue is already closed. I close this PR. Thank for review.

**Pull request closed without merging**